File: Data Structures & Algorithms/is-anagram/submission-3.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> v(26, 0);
        for(const char& c : s) v[c - 'a']++;
        for(const char& c : t) v[c - 'a']--;
        cout << v['x' - 'a'] << endl;
        for(const int& e : v) if (e != 0) return false;
        return true;
    }
};

File: Data Structures & Algorithms/minimum-window-with-characters/submission-3.py
class Solution:
    def minWindow(self, s: str, t: str) -> str:
        if t == "": return ""
        
        cntT, window = {}, {}

        for c in t:
            cntT[c] = cntT.get(c, 0) + 1
        
        windowCommonCharSize, tLen = 0, len(cntT)
        l = 0
        res, resLen = [-1, -1], float("inf")
        for r in range(len(s)):
            c = s[r]
            window[c] = window.get(c, 0) + 1
            if c in cntT and window[c] == cntT[c]:
                windowCommonCharSize += 1
            
            while windowCommonCharSize == tLen:
                if (r - l + 1) < resLen:
                    res = [l, r]
                    resLen = r - l + 1
                window[s[l]] -= 1
                if s[l] in cntT and window[s[l]] < cntT[s[l]]:
                    windowCommonCharSize -= 1
                l += 1
        return s[res[0]: res[1]+1] if resLen != float("inf") else ""